### PR TITLE
UI Toggle & API for SuperAdminFirstUserOnly server config

### PR DESCRIPTION
**Pull request by alexpaxton**
Connect #2342 

### To-Do
- [x] Wire up the toggle
- [x] Add `/chronograf/v1/config` & `/chronograf/v1/config/auth` to links
- [x] Create GET & UPDATE service functions & API for `superAdminFirstUserOnly` bool
- [x] Deprecate the CLI flag
- [x] Rename & flip logic of superAdminFirstUserOnly to superAdminNewUsers for clarity

### The Problem
- We don't want to introduce the `--superadmin-first-user-only` CLI flag, and would rather make this a config option that is togglable via the UI.
- It was semantically weird to have SuperAdminFirstUserOnly be off in the UI out of the box.

### The Solution
- Create a "Config" table beneath the Organizations Table
  - SuperAdmins can already see this page
  - If/when we have more settings we can find a better home for this
- Add API and Bolt store for Config GET and UPDATE
- Change SuperAdminFirstUserOnly to SuperAdminNewUsers, and flip the default state
- Deprecate the CLI flag for this

### Final preview
screen shot 2017-12-13 at 2 54 52 pm

### Old preview
screen shot 2017-11-10 at 6 25 32 pm
** The last 2 options not included in this PR

### CODE REVIEW

**Comment by jaredscheib** [comment]
Regarding the data shape and API, @goller is it possible that there are non-superadmin settings that we may wish to put at `chronograf/v1/application`? I'm wondering if maybe these auth settings 
should go under `/application/auth`, for instance, in case we end up storing global settings, such as color scheme, under `/application`? Or would those go under the `/user/:id/settings`? And what about the key name `settings` instead of `application`?

**Comment by goller** [comment]
> Regarding the data shape and API, @goller is it possible that there are non-superadmin settings that we may wish to put at chronograf/v1/application? 

I think not.  These settings are related to how the application should be configured as a process. I'm more in favor of `/chronograf/v1/config` following @nathanielc 's kapacitor config style.

> I'm wondering if maybe these auth settings should go under /application/auth,

I'm all about scoped REST resources.  Auth seems like it's own resource scoped under config like
`/chronograf/v1/config/auth/:id`

> for instance, in case we end up storing global settings, such as color scheme, under /application? 

Color-schemes, white-labeling, and such seems like an application-wide configuration option and should likely be it's own resource that may be modified... perhaps `/chronograf/v1/config/colors`.

I'd have to have a specific feature to suss out the specific resource name, but, that's the general idea.

> Or would those go under the /user/:id/settings? 

Perhaps having user settings works.  I'm not sure yet what those would be, but, yes.  However, if you mean could a user have personalized color settings, (high-contrast for accessibility, etc) yes, we'd have some sort of resource that does it.   If colors could be changed universally for a person, we'd allow a PATCH to /me/views, or some such.

However, in designing APIs I really, really try to break them down to the specific concept rather than catch-all terms like settings.  Otherwise those particular resources become very, very heterogeneous and difficult to update and use.



> And what about the key name settings instead of application?

I'm ok with settings or config.  I don't really like application.  Quick survey of other APIs... I see `customization`, `settings`, `config`

**Comment by jaredscheib** [comment]
Thanks @goller! And on that note, here's another configuration issue that was just opened up: https://github.com/influxdata/chronograf/issues/2551. Look out for a PR review request soon!

**Comment by jaredscheib** [comment]
Done, @lukevmorris.

File: server/me.go [hunk 1]
```diff
@@ -319,7 +319,12 @@ func (s *Service) newUsersAreSuperAdmin() bool {
 	if s.firstUser() {
 		return true
```
Comment (ebb-tide) [inline server/me.go]: if cfg.Auth.SuperAdminNewUsers is true by default- is this still needed ? -luke+deniz

File: ui/src/admin/components/chronograf/OrganizationsTable.js [hunk 2]
```diff
@@ -89,13 +98,35 @@ class OrganizationsTable extends Component {
...
+                  <th />
+                </tr>
+              </thead>
+              <tbody>
+                <tr>
+                  <td style={{width: 70}}>
+                    <SlideToggle
+                      size="xs"
+                      active={superAdminNewUsers}
+                      onToggle={this.handleChangeAuthConfigSuperAdminNewUsers}
```
Comment (ebb-tide) [inline ui/src/admin/components/chronograf/OrganizationsTable.js]: Suggestion to consider:
``` js 
onToggle={this.props.handleChangeAuthConfig('superAdminNewUsers')}
```
delete handleChangeAuthConfigSuperAdminNewUsers function:
and in OrganizationsPage.js replace handleUpdateAuthConfig with: 

```js
handleChangeAuthConfig=(fieldName)=>(updatedValue)=>{ 
  const { 
    actionsConfig: {updateAuthConfigAsync},
    authConfig,
    links,
  } = this.props
  const updatedAuthConfig= {...authConfig, [fieldName]:updatedValue}
  updateAuthConfigAsync(links.config.auth, authConfig, updatedAuthConfig)
}
```

File: ui/src/shared/reducers/config.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const initialState = {
+  links: {},
+  auth: {},
+}
+
+const config = (state = initialState, action) => {
```
Comment (ebb-tide) [inline ui/src/shared/reducers/config.js]: "this looks great. I like this reducer" -luke

File: ui/src/utils/ajax.js [hunk 4]
```diff
@@ -19,6 +27,7 @@ const generateResponseWithLinks = (response, newLinks) => {
     users,
     organizations,
     meLink,
+    config,
```
Comment (ebb-tide) [inline ui/src/utils/ajax.js]: "great!"-luke

File: server/me.go (hunk 1 above)
Comment (jaredscheib) [reply server/me.go]: @desa ^^?

File: ui/src/admin/components/chronograf/OrganizationsTable.js (hunk 2 above)
Comment (jaredscheib) [reply ui/src/admin/components/chronograf/OrganizationsTable.js]: good suggestion - thanks!

File: server/me.go (hunk 1 above)
Comment (desa) [reply server/me.go]: It's not needed if the default is true, but there's nothing else in the application that dictates that it must be true. So for that reason, I'd like to keep it there.

We've talked about possibly allowing users to override default values via CLI and this case could possibly happen then.

Comment (jaredscheib) [reply server/me.go]: Could we comment that in the code? Then we're good to merge, I think

**Comment by jaredscheib** [comment]
This is ready to merge once circle passes.

**Pull request merged**